Don't expect dependencies to be frameworks unless LOVE_MACOSX_USE_FRAMEWORKS
is defined (and we don't define it).
https://github.com/love2d/love/commit/eb4fdf3bbe8a0ca38b2f2eaaf14e29e46ee27f94
--- src/modules/audio/openal/Audio.h.orig	2012-04-02 15:31:32.000000000 -0500
+++ src/modules/audio/openal/Audio.h	2024-05-08 00:44:07.000000000 -0500
@@ -37,7 +37,7 @@
 #include <thread/threads.h>
 
 // OpenAL
-#ifdef LOVE_MACOSX
+#ifdef LOVE_MACOSX_USE_FRAMEWORKS
 #include <OpenAL/alc.h>
 #include <OpenAL/al.h>
 #else
--- src/modules/audio/openal/Pool.h.orig	2012-04-02 15:31:32.000000000 -0500
+++ src/modules/audio/openal/Pool.h	2024-05-08 00:43:58.000000000 -0500
@@ -33,7 +33,7 @@
 #include <thread/threads.h>
 
 // OpenAL
-#ifdef LOVE_MACOSX
+#ifdef LOVE_MACOSX_USE_FRAMEWORKS
 #include <OpenAL/alc.h>
 #include <OpenAL/al.h>
 #else
--- src/modules/audio/openal/Source.h.orig	2012-04-02 15:31:32.000000000 -0500
+++ src/modules/audio/openal/Source.h	2024-05-08 00:44:12.000000000 -0500
@@ -29,7 +29,7 @@
 #include <sound/Decoder.h>
 
 // OpenAL
-#ifdef LOVE_MACOSX
+#ifdef LOVE_MACOSX_USE_FRAMEWORKS
 #include <OpenAL/alc.h>
 #include <OpenAL/al.h>
 #else
--- src/modules/filesystem/physfs/File.h.orig	2012-04-02 15:31:32.000000000 -0500
+++ src/modules/filesystem/physfs/File.h	2024-05-07 21:05:48.000000000 -0500
@@ -25,7 +25,7 @@
 #include <filesystem/File.h>
 
 // PhysFS
-#ifdef LOVE_MACOSX // wacky Mac behavior means different #include syntax!
+#ifdef LOVE_MACOSX_USE_FRAMEWORKS
 #include <physfs/physfs.h>
 #else
 #include <physfs.h>
--- src/modules/font/freetype/Font.h.orig	2012-04-02 15:31:32.000000000 -0500
+++ src/modules/font/freetype/Font.h	2024-05-07 21:10:48.000000000 -0500
@@ -25,7 +25,7 @@
 #include <font/Font.h>
 
 // FreeType2
-#ifdef LOVE_MACOSX
+#ifdef LOVE_MACOSX_USE_FRAMEWORKS
 #include <freetype/ft2build.h>
 #else
 #include <ft2build.h>
--- src/modules/sound/lullaby/Mpg123Decoder.h.orig	2012-04-02 15:31:32.000000000 -0500
+++ src/modules/sound/lullaby/Mpg123Decoder.h	2024-05-07 22:37:34.000000000 -0500
@@ -26,7 +26,7 @@
 #include "Decoder.h"
 
 // libmpg123
-#ifdef LOVE_MACOSX
+#ifdef LOVE_MACOSX_USE_FRAMEWORKS
 #include <mpg123/mpg123.h>
 #else
 #include <mpg123.h>
